WatsonTcp/WatsonTcpClient.cs

```diff
@@ -886,8 +886,15 @@ private async Task DataReceiver()
 
             Connected = false;
 
-            _Settings.Logger?.Invoke(Severity.Debug, _Header + "data receiver terminated for " + _ServerIp + ":" + _ServerPort);
-            _Events.HandleServerDisconnected(this, new DisconnectionEventArgs((_ServerIp + ":" + _ServerPort), reason));
+            if (_Settings != null)
+            {
+                _Settings.Logger?.Invoke(Severity.Debug, _Header + "data receiver terminated for " + _ServerIp + ":" + _ServerPort);

```

Comment:
This is excellent @ozrecsec - will be pushing an update shortly.  Instead of using this syntax, I've changed to use null conditionals on both ```_Settings``` and ```_Logger```:
```
_Settings?.Logger?...
```